src/audio/module_adapter/module/generic.c

```diff
@@ -114,7 +121,13 @@ int module_init(struct processing_module *mod)
 	mod->priv.resources.rsrc_mngr = k_current_get();
 #endif
 	/* Now we can proceed with module specific initialization */
-	ret = interface->init(mod);
+#if CONFIG_IPC_MAJOR_4
+	if (mod->dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP)
```

Comment:
Is this safe for the proxy-dp implementation?

Reply:
Definitely not. This should be wrapped in an `#ifdef`, just like the other calls to this function.